tests/parameter_handler/parameter_handler_subsection_exists.cc

```diff
@@ -0,0 +1,118 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2003 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// Test ParameterHandler::subsection_path_exists().
+
+#include <deal.II/base/parameter_handler.h>
+
+#include <iomanip>
+
+#include "../tests.h"
+
+int
+main()
+{
+  try
```

Comment:
no need to catch exceptions.

Reply:
Many of the functions used therein (`declare_entry`, `print_parameters`) may possibly throw exceptions, do you think it is redundant to keep the `try/catch` block?

Reply:
The default exception handler is good enough to give meaningful error messages. We (usually) only catch in tests if we want to check that the correct exception is thrown.